tds-dll-common/src/main/java/tds/dll/common/performance/caching/impl/JedisSentinelConnectionFactory.java

```diff
@@ -0,0 +1,181 @@
+package tds.dll.common.performance.caching.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.redis.RedisConnectionFailureException;
+import org.springframework.data.redis.connection.RedisConnection;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConverters;
+import org.springframework.util.Assert;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.JedisSentinelPool;
+import redis.clients.jedis.Protocol;
+import redis.clients.util.Pool;
+
+import java.util.Set;
+
+import static com.google.common.collect.Sets.newHashSet;
+
+/**
+ * This class is a copy of Spring's JedisConnectionFactory with support for a JedisSentinelPool.
```

Comment:
Literally a copy.  I would have rather extended the old version, but it had hard-coded a JedisPool as the pool implementation rather than a Pool<Jedis> which means I couldn't use a JedisSentinelPool